src/components/BottomBar/SkyBrowser/SkyBrowserSettings.jsx

```diff
@@ -22,67 +20,38 @@ import {
  } from '../../../api/keys';
 import styles from './SkyBrowserSettings.scss';
 
-class SkyBrowserSettings extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showExpandedCopies: false,
-      showExpandedSettings: false,
-      generalSettingsExpanded: false,
-      precisionLow: 2,
-      precisionHigh: 10,
-      newPosition: [0, 0, -2],
-      noOfCopies: 1,
-      showCopiesInfo: false,
-    };
-    this.valueToColor = this.valueToColor.bind(this);
-    this.onColorPickerChange = this.onColorPickerChange.bind(this);
-    this.toggleFaceCamera = this.toggleFaceCamera.bind(this);
-    this.toggleRadiusAzimuthElevation = this.toggleRadiusAzimuthElevation.bind(this);
-    this.setRef = this.setRef.bind(this);
-    this.setExpandedCopies = this.setExpandedCopies.bind(this);
-    this.createDisplayCopiesSection = this.createDisplayCopiesSection.bind(this);
-    this.showExpandedCopySettings = this.showExpandedCopySettings.bind(this);
-    this.setExpandedGeneralSettings = this.setExpandedGeneralSettings.bind(this);
-  }
+function SkyBrowserSettings({
+   luaApi, selectedBrowserId, browser, setBorderRadius
+}) {
+  const [showExpandedCopies, setShowExpandedCopies] = React.useState(false);
+  const [showExpandedSettings, setShowExpandedSettings] = React.useState(false);
+  const [generalSettingsExpanded, setGeneralSettingsExpanded] = React.useState(false);
+  const [newPosition, setNewPosition] = React.useState([0, 0, -2]);
+  const [noOfCopies, setNoOfCopies] = React.useState(1);
 
-  componentDidMount() {
-    const { browser } = this.props;
-    this.setState({
-      newPosition : browser.isUsingRae ? [2, 0, 0] : [0, 0, -2]
-    });
-  }
+  const precisionLow = 2;
+  const precisionHigh = 10;
 
-  setRef(what) {
-    return (element) => {
-      this[what] = element;
-    };
-  }
-
-  setExpandedCopies(expanded) {
-    this.setState({ showExpandedCopies: expanded });
-  }
+  React.useEffect(() => {
+    setNewPosition(browser.isUsingRae ? [2, 0, 0] : [0, 0, -2]);
+  }, []);
 
-  setExpandedGeneralSettings(expanded) {
-    this.setState({ generalSettingsExpanded: expanded });
+  if (!browser) {
+    return null;
   }
 
-  toggleFaceCamera() {
-    const { luaApi, browser } = this.props;
+  function toggleFaceCamera() {
     const uriBrowser = `ScreenSpace.${browser.id}.FaceCamera`;
     luaApi.setPropertyValueSingle(uriBrowser, !browser.isFacingCamera);
   }
 
-  toggleRadiusAzimuthElevation() {
-    const { luaApi, browser } = this.props;
+  function toggleRadiusAzimuthElevation() {
     const uriBrowser = `ScreenSpace.${browser.id}.UseRadiusAzimuthElevation`;
     luaApi.setPropertyValueSingle(uriBrowser, !browser.isUsingRae);
-    this.setState({
-      newPosition : !browser.isUsingRae ? [2, 0, 0] : [0, 0, -2]
-    });
+    setNewPosition(browser.isUsingRae ? [0, 0, -2] : [2, 0, 0]);
```

Comment:
This is the seocnd time these exact vectors are used. Can they be variables with meningful names?